superset-frontend/cypress-base/cypress/integration/explore/control.test.ts

```diff
@@ -233,8 +233,158 @@ describe('Time range filter', () => {
+          const secondRange = $range.text();
+          expect(secondRange).to.not.equal(firstRange);
+        });
+      });
+  });
+  it('Chosen last time filters are applied after change', () => {
+    const formData = {
+      ...FORM_DATA_DEFAULTS,
+      metrics: [NUM_METRIC],
+      viz_type: 'line',
+      time_range: '100 years ago : now',
+    };
+
+    cy.visitChartByParams(JSON.stringify(formData));
+    cy.verifySliceSuccess({ waitAlias: '@postJson' });
+
+    cy.get('[data-test=time-range-trigger]')
+      .click()
+      .then($range => {
+        const firstRange = $range.text();
+        cy.get('.ant-popover-inner-content').within(() => {
+          cy.contains('Advanced').click();
+          cy.contains('Last').click();
+          cy.get('.ant-radio-wrapper').within(() => {
+            cy.contains('last week').click();
+          });
+        });
+        cy.get('[data-test=apply-button]').click();
+        cy.get('[data-test=time-range-modal]').should('not.exist');
+        cy.get('[data-test=time-range-trigger]').then($range => {
+          const secondRange = $range.text();
+          expect(secondRange).to.not.equal(firstRange);
+        });
+        cy.get('[data-test=time-range-trigger]')
+          .click()
+          .then($range => {
+            const firstRange = $range.text();
+            cy.get('.ant-popover-inner-content').within(() => {
+              cy.contains('Last').click();
+              cy.get('.ant-radio-wrapper').within(() => {
+                cy.contains('last year').click();
+              });
+            });
+            cy.get('[data-test=apply-button]').click();
+            cy.get('[data-test=time-range-modal]').should('not.exist');
+            cy.get('[data-test=time-range-trigger]').then($range => {
+              const secondRange = $range.text();
+              expect(secondRange).to.not.equal(firstRange);
+            });
+          });
+      });
+  });
+  it('Changes through range types filters are applied after change', () => {
+    const formData = {
+      ...FORM_DATA_DEFAULTS,
+      metrics: [NUM_METRIC],
+      viz_type: 'line',
+      time_range: '100 years ago : now',
+    };
+
+    cy.visitChartByParams(JSON.stringify(formData));
+    cy.verifySliceSuccess({ waitAlias: '@postJson' });
+    cy.get('[data-test=time-range-trigger]')
+      .click()
+      .then(() => {
+        cy.get('.ant-popover-inner-content').within(() => {
+          cy.contains('Advanced').click({ force: true });
+          cy.contains('Custom').click({ force: true });
+          cy.get('.ant-input-number-input').clear().type('35{enter}');
+        });
+
+        cy.get('[data-test=apply-button]').click();
+        cy.get('[data-test=time-range-modal]').should('not.exist');
+        cy.get('[data-test=time-range-trigger]').then($range => {
+          const customRange = $range.text();
+          cy.get('[data-test=time-range-trigger]')
+            .click()
+            .then(() => {
```

Comment:
Is the then wrapper necessary? I thought a Cypress selector will just keep retrying until it finds the elements or timeout.

You can probably reselect the same trigger element to avoid nesting?

Reply:
I do not see an option to avoid nesting -> I need to have an access to text value of range when I go to another option to verify. I cannot export this to variable.